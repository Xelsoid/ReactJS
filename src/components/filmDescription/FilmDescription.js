import PropTypes from "prop-types";
import React from 'react';
import {connect} from "react-redux";

import './filmDescription.scss';

const FilmDescription = ({selectedFilm}) => {
  if(!selectedFilm){return null}

  const {poster_path, title, tagline, release_date, runtime, overview} = selectedFilm;
  return (
    <div className="film-description clearfix">
      <div className="film-description__image-wrapper">
        <img alt={title} className="film-description__image" src={poster_path} />
      </div>
      <div className="film-description__info-wrapper">
        <h2 className="film-description__header">{title}</h2>
        <strong className="film-description__useful-info display-block">{tagline}</strong>
        <span>
          {`Year: ${new Date(release_date).getFullYear()} `}
        </span>
        <span>
          {`${runtime} min`}
        </span>
        <p className="film-description__narration">
          {overview}
        </p>
      </div>
    </div>
  );
};

function mapStateToProps(state) {
  return {
    selectedFilm: state.selectedFilm
  };
}

FilmDescription.defaultProps = {
  selectedFilm: {},
};

FilmDescription.propTypes = {
  selectedFilm: PropTypes.instanceOf(Object),
};

export default connect(mapStateToProps)(FilmDescription);
