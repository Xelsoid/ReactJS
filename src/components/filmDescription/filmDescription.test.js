import React from 'react';
import enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import FilmDescription from './FilmDescription';

enzyme.configure({ adapter: new Adapter() });

it('renders correctly', () => {
    const component = shallow (
      <FilmDescription />
    );
    expect(component).toMatchSnapshot();
});

