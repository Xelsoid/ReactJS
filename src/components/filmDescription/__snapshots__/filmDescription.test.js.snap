// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDescription renders correctly 1`] = `
<div
  className="film-description clearfix"
>
  <div
    className="film-description__image-wrapper"
  >
    <img
      className="film-description__image"
    />
  </div>
  <div
    className="film-description__info-wrapper"
  >
    <h2
      className="film-description__header"
    />
    <strong
      className="film-description__useful-info display-block"
    />
    <span>
      Year: NaN 
    </span>
    <span>
      undefined min
    </span>
    <p
      className="film-description__narration"
    />
  </div>
</div>
`;
